File: app/src/main/java/com/example/test/data/users/UserDao.java
package com.example.test.data.users;

import androidx.lifecycle.LiveData;
import androidx.paging.PagingSource;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface UserDao {

    @Query("SELECT * FROM users ORDER BY login COLLATE NOCASE ASC")
    PagingSource<Integer, UserData> getSource();

    @Query("SELECT * FROM users ORDER BY login COLLATE NOCASE ASC")
    LiveData<List<UserData>> getUser();

    @Insert
    void insert(List<UserData> data);

    @Insert
    void insert(UserData data);

    @Query("DELETE FROM users")
    void delete();
}

File: app/src/main/java/com/example/test/data/users/UserDao.kt
package com.example.test.data.users

import androidx.paging.PagingSource
import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface UserDao {
    @Query("SELECT * FROM users ORDER BY login COLLATE NOCASE ASC")
    fun getSource(): PagingSource<Int, UserData>

    @Query("SELECT * FROM users ORDER BY login COLLATE NOCASE ASC")
    fun getUser(): Flow<List<UserData>>

    @Insert
    fun insert(data: List<UserData>)

    @Insert
    suspend fun insert(data: UserData)

    @Query("DELETE FROM users")
    suspend fun delete()
}